src/test/java/wooteco/subway/service/LineServiceTest.java

```diff
@@ -0,0 +1,150 @@
+package wooteco.subway.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.BDDMockito.given;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.boot.test.context.SpringBootTest;
+import wooteco.subway.dao.LineDao;
+import wooteco.subway.domain.Line;
+import wooteco.subway.dto.LineRequest;
+import wooteco.subway.dto.LineResponse;
+
+@SpringBootTest
+class LineServiceTest {
+    @Mock
+    LineDao lineDao;
+
+    @InjectMocks
+    LineService lineService;
```

Comment:
mock을 활용해주셨군요!

`@Mock`과 `@MockBean`은 어떤 차이가 있을까요?
이 테스트에는 `@SpringBootTest`가 필요할까요?
[둘의 차이점](https://www.baeldung.com/java-spring-mockito-mock-mockbean)에 대해서 학습해볼까요?

Reply:
MockBean은 Mock과 달리 스프링이 제공합니다! 
Spring Application Context에 추가되기 때문에 InjectMock이 아니라 Autowired로 접근이 가능합니다. 
또한 @SpringBootTest는 의존성을 제공하기 때문에 해당 어노테이션이 없다면 NullPointException이 발생하게 됩니다.

Reply:
```suggestion
@ExtendWith(MockitoExtension.class)
class LineServiceTest {
    @Mock
    LineDao lineDao;

    @InjectMocks
    LineService lineService;
```
말씀해주신대로 MockBean, SpringBootTest는 모두 스프링(부트)과 관련이 있는 어노테이션입니다.
반면에 Mock, InjectMocks는 Mockito와 관련이 있죠. 스프링과는 관련이 없습니다.

Mock은 [슬라이스 테스트](https://tecoble.techcourse.co.kr/post/2021-05-18-slice-test/)를 편하게 도와주는 역할을 하는데요, 슬라이스 테스트를 할 때는 스프링 관련 의존성을 불러올 필요가 없습니다. 즉, SpringBootTest가 필요가 없어지겠죠.
NPE는 그럼 왜 발생할까요? 단순히 `@Mock`, `@InjectMocks`를 붙여주는 것만으로는 mockito가 동작할 수 없습니다. 몇 가지 세팅이 필요한데요, 이를 해주는게 `MockitoExtension`입니다. `@SpringBootTest`로 모든 의존성을 불러올 필요없이 필요한 객체만 mocking하는 방식을 사용해보면 좋을 것 같습니다!

참고로 mockito setting을 불러오지 않는다면 다음과 같이 구현할 수도 있어요~
```java
class LineServiceTest {
    LineDao lineDao;

    LineService lineService;

    @BeforeEach
    public void setUp() {
        lineDao = Mockito.mock(LineDao.class);
        lineService = new LineService(lineDao);
    }
```

Reply:
와 MockitoExtension과 setUp 모두 학습해보지 못했던 부분이네요ㅠㅠ! 알려주셔서 너무 감사합니다. 후자의 방식을 이용하여 리팩토링 해보았습니다! 😃